pkg/controller/cluster/main-controller.go

```diff
@@ -1141,6 +1141,11 @@ func (c *Controller) syncHandler(key string) error {
 		return err
 	}
 
+	if err := c.createBuckets(ctx, tenant, tenantConfiguration); err != nil {
```

Comment:
I think you should introduce a field in `.Status` to acknowledge that the buckets have been created already, else you will be calling this every single time the tenant gets sync which happens often, in the case the bucket was deleted inside MinIO the operator would end up recreating it, over and over

Reply:
done